### ISSUE TITLE
Google Analytics on Laravel Voyager Admin

### ISSUE THREAD

**Issue by shrihub** [thedevdojo/voyager#2216]
- Laravel Version: 5.4.36
- Voyager Version: v1.0.7
- PHP Version: 5.6.30
- Database Driver & Version: 

### Description:
After adding google analytics client ID in Voyager, sometimes you may not get the analytics report right away in the voyager dashboard page; Instead it will show the below text.

You are logged in as: ***@***.com

### Steps To Reproduce:
1. Create a google analytics client ID under https://console.developers.google.com
2. Return to Voyager admin -> Settings -> Add the client ID under admin section and save.
3. Go to Voyager dashboard where you may only see the text "You are logged in as: xxxx@email.com"

### Solution:  Enable google analytics API after creating your client ID.
1. Go to https://console.developers.google.com
2. Select Library
3. Search for the keyword 'analytics'
4. Click on 'Analytics API'
5. Click on Enable

![screen shot 2017-11-11 at 11 26 21 am](https://user-images.githubusercontent.com/4091734/32699433-9f2cf2b4-c7db-11e7-94ee-5e70528e7cc4.png)
![screen shot 2017-11-11 at 11 26 36 am](https://user-images.githubusercontent.com/4091734/32699434-9f8c4070-c7db-11e7-93cc-4883a2dcb607.png)
![screen shot 2017-11-11 at 11 28 32 am](https://user-images.githubusercontent.com/4091734/32699435-9fcd2dec-c7db-11e7-91f9-358460fb8374.png)

**Comment by MicrowaveDev** [comment]
Same problem

**Comment by mattrabe** [comment]
Thank you for posting this

**Comment by chrisdicarlo** [comment]
Still seems to be an issue?

**Comment by mattrabe** [comment]
I was having this problem previously, but not longer am thanks to this post. After enabling the API in the Google Library **I also had to make sure to log in to google using the correct account** (I have multiple google accounts and only one of them had access to the analytics data I was trying to access), but after that it works well for me.

**Comment by harunthuo** [comment]
this worked for me. thankyou

**Comment by ghost** [comment]
same problem

**Comment by fletch3555** [comment]
Guys, please stop commenting "Same problem".  This issue isn't about a problem (at least not one within Voyager's control).  It's about adding documentation to assist devs with getting past this step.

**Comment by ghost** [comment]
I think the problem happen because http must be https in xampp

**Comment by mukhlisa96** [comment]
I guess you should install an analytics tool on your page.
go to **https://analytics.google.com/analytics** and register your website. then copy the code like  
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    ```
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-***********"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-***********');
</script>
```**
</head>
and past it inside head tag as the first element.

**Comment by thelesson** [comment]
If someone encountered a problem by following these steps. Delete the OAuth ID and create again after enabling the Analytics API.
I enabled the api after creating the OAuth ID and it didn't work, only after deleting and enabling it again that I managed to make it work

**Comment by alkhachatryan** [comment]
Same problem, 2021.

@fletch3555 you say it's not a Voyager problem, but I did as Voyager docs said.

**Comment by arsengoian** [comment]
@alkhachatryan see https://github.com/the-control-group/voyager/issues/5495

It seems to be a new issue. Did you find a solution?

**Comment by umutcanli** [comment]
same problem from 2022

**Comment by EslamMuhammedSaeed** [comment]
Same problem

**Comment by arsengoian** [comment]
Guys, the solution is to build a server-side integration with GA 4 Data API.

see
https://packagist.org/packages/google/analytics-data
https://developers.google.com/analytics/devguides/reporting/data/v1

If only anyone coult imlpement this and make a PR. We did this, but with a third-party proprietary charting library on front-end, so we can't unfortunately offer our solution.

This could literally be done in a day 🤔

I would be happy to assist to anyone working on such a PR

**Comment by Andre-ADPC** [comment]
Hi, off on a tangent here regarding GA: Google is facing more and more resistance from the world outside the US in regards to stiff legal sanctions and fines imposed on them at governmental levels in the EU recently. Yes it is free to use for devs, but at what hidden cost? 
Ref A: [European rulings on the use of Google Analytics and how it may affect your business](https://www.dataprotectionreport.com/2022/02/european-rulings-on-the-use-of-google-analytics-and-how-it-may-affect-your-business/)
Ref B: [EU declares Google Analytics illegal: Here’s why](https://techstory.in/eu-declares-google-analytics-illegal-heres-why/)

What about considering giving devs an option between something like GA and Fathom for example. Yes, there are others, but Fathom seems quite adamant that they out-perform GA substantially, and are rooting for the business customers' and in return their end user's privacy. Google is in it for one purpose, "Get their data and leverage it into cash", period. 

GitHub is one of their main customers too. Ironic? Probably not.

The choice is thus offered as an option at solution presentation time and the developer can then implement the option the customer decides on.

Here is a reference article, one of many, that discusses why GA is losing and is going to lose more track: [Fathom Analytics vs Google Analytics](https://pauljardine.co.uk/blog/fathom-analytics-vs-google-analytics/)

The most interesting thing is that most of you can add it to Voyager yourselves. Fathom states that it is a mere "one-liner" in your head section.

Their words:
"**_To start using Fathom Analytics, you simply have to add a single line of code to the head section of your website (where “ABCDEFG” is your actual Site ID):_**"

`<script src="https://cdn.usefathom.com/script.js" data-site="ABCDEFG" defer></script>`

"**_However, we recommend using [a custom domain](https://usefathom.com/docs/script/custom-domains), as those bypass ad-blockers and show you 100% of your traffic._**"

There are several packages available on [Packagist](https://packagist.org/packages/google/analytics-data?query=fathom) that should make your lives easier when implementing it in more customised approaches too.

This is just an observation from a guy that has learned to Never Say Never.
